lib/controllers/european.controller.ts

```diff
@@ -75,16 +84,18 @@ export class EuropeanController extends SessionController {
           password: this.userConfig.password,
         },
         cookieJar,
+      }).catch(err => {
+        logger.debug(`Login failed: ${err}`);
+        Promise.reject(`Login failed: ${err}`);
       });
 
-      const queryString = authCodeResponse.body.redirectUrl.split('?')[1];
-      const urlParams = new URLSearchParams(queryString);
-      const authToken = urlParams.get('code');
-
-      if (authToken !== null) {
-        this.session.refreshToken = authToken;
-      } else {
-        throw new Error('@EuropeControllerLogin: AuthCode was not found');
+      if(authCodeResponse){
+        const regexMatch = /.*code=(\w*)&.*/g.exec(authCodeResponse.body.redirectUrl);
```

Comment:
Does this regex work with codes that have dashes? I ran into an issue that codes were coming back with dashes and getting failures.

Reply:
Have yet to see a dash in the codes, but we have to deal with that then